dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1722,6 +1726,40 @@ async fn cmd_nexus_blueprints_target_set(
     args: &BlueprintTargetSetArgs,
     _destruction_token: DestructiveOperationToken,
 ) -> Result<(), anyhow::Error> {
+    let found_enabled = if args.diff
+        || matches!(args.enabled, BlueprintTargetSetEnabled::Inherit)
```

Comment:
This (and the unwrap) seem kinda awkward. Would it be too clever to stick "get the current target" in a tokio `OnceCell`? Something like

```rust
    // Helper to only fetch the current target once. We may need it immediately
    // if `args.diff` is true, or later if `args.enabled` is "inherit" (or
    // both).
    let current_target = OnceCell::new();
    let get_current_target = || async {
        current_target
            .get_or_try_init(|| client.blueprint_target_view())
            .await
            .context("failed to fetch current target blueprint")
    };
```

at which point we could drop the check for `args.enabled` entirely here:

```diff
-    let found_enabled = if args.diff
-        || matches!(args.enabled, BlueprintTargetSetEnabled::Inherit)
-    {
-        let current_target = client
-            .blueprint_target_view()
-            .await
-            .context("failed to fetch current target blueprint")?
-            .into_inner();
-
     if args.diff {
+        let current_target = get_current_target().await?;
         let blueprint1 = client
             .blueprint_view(&current_target.target_id)
             .await
@@ -1755,11 +1748,6 @@ async fn cmd_nexus_blueprints_target_set(
         prompt.read_and_validate("y/N", "y")?;
     }

-        Some(current_target.enabled)
-    } else {
-        None
-    };
-
     let enabled = match args.enabled {
         BlueprintTargetSetEnabled::Enabled => true,
         BlueprintTargetSetEnabled::Disabled => false,
@@ -1772,9 +1760,9 @@ async fn cmd_nexus_blueprints_target_set(
         // operator. (In the case of the current target blueprint being changed
         // entirely, that will result in a failure to set the current target
         // below, because its parent will no longer be the current target.)
-        //
-        // unwrap(): this is always `Some` due to the branch above.
-        BlueprintTargetSetEnabled::Inherit => found_enabled.unwrap(),
+        BlueprintTargetSetEnabled::Inherit => {
+            get_current_target().await?.enabled
+        }

Reply:
Agreed.  I tried and failed to do this before.  I'll take a look at that diff and see if I can make it work.

Reply:
Great!  I like this much better.  Done in f9beb4c.